domain/consensus/model/database.go

```diff
@@ -1,14 +1,96 @@
 package model
 
-import "github.com/kaspanet/kaspad/domain/consensus/model/externalapi"
+// DBCursor iterates over database entries given some bucket.
+type DBCursor interface {
+	// Next moves the iterator to the next key/value pair. It returns whether the
+	// iterator is exhausted. Panics if the cursor is closed.
+	Next() bool
 
-// DBContextProxy defines a proxy over domain data access
-type DBContextProxy interface {
-	FetchBlockRelation(blockHash *externalapi.DomainHash) (*BlockRelations, error)
+	// First moves the iterator to the first key/value pair. It returns false if
+	// such a pair does not exist. Panics if the cursor is closed.
+	First() bool
+
+	// Seek moves the iterator to the first key/value pair whose key is greater
+	// than or equal to the given key. It returns ErrNotFound if such pair does not
+	// exist.
+	Seek(key DBKey) error
+
+	// Key returns the key of the current key/value pair, or ErrNotFound if done.
+	// The caller should not modify the contents of the returned key, and
+	// its contents may change on the next call to Next.
+	Key() (DBKey, error)
+
+	// Value returns the value of the current key/value pair, or ErrNotFound if done.
+	// The caller should not modify the contents of the returned slice, and its
+	// contents may change on the next call to Next.
+	Value() ([]byte, error)
+
+	// Close releases associated resources.
+	Close() error
+}
+
+// DBReader defines a proxy over domain data access
+type DBReader interface {
+	// Get gets the value for the given key. It returns
+	// ErrNotFound if the given key does not exist.
+	Get(key DBKey) ([]byte, error)
+
+	// Has returns true if the database does contains the
+	// given key.
+	Has(key DBKey) (bool, error)
+
+	// Delete deletes the value for the given key. Will not
+	// return an error if the key doesn't exist.
+	Delete(key DBKey) error
+
+	// Cursor begins a new cursor over the given bucket.
+	Cursor(bucket DBBucket) (DBCursor, error)
 }
 
-// DBTxProxy is a proxy over domain data
+// DBTransaction is a proxy over domain data
 // access that requires an open database transaction
-type DBTxProxy interface {
-	StoreBlockRelation(blockHash *externalapi.DomainHash, blockRelationData *BlockRelations) error
+type DBTransaction interface {
+	// Put sets the value for the given key. It overwrites
+	// any previous value for that key.
+	Put(key []byte, value []byte) error
+
+	// Delete deletes the value for the given key. Will not
+	// return an error if the key doesn't exist.
+	Delete(key []byte) error
```

Comment:
Please extract Put and Delete to an interface called DBWriter and have DBTransaction inherit from it.